### DISCUSSION TITLE
Optimize (-O -O1 -O2...) with PlatformIO

### DISCUSSION THREAD

**Discussion by LazaroFilm** [earlephilhower/arduino-pico#1721] [Q&A] [answered]
I can't find the optimization option in PlatformIO. I see it in Arduino IDE's menu but I can't find where it would be in VS Code's PlatformIO...

**Comment by maxgerhardt** [comment, marked as answer]
PlatformIO applies a default optimization of `-Os`.  As documented in [build_flags](https://docs.platformio.org/en/latest/projectconf/sections/env/options/build/build_flags.html) and [build_unflags](https://docs.platformio.org/en/latest/projectconf/sections/env/options/build/build_unflags.html), you can throw the default one out and insert your own in the `platformio.ini` like so:
```ini
; throw out default optimize for size
build_unflags = -Os
; let's make this slow on purpose because why not ^_^
build_flags = -O0
```
You can always verify that your wanted flags are applied by looking at the full compiler invocations as shown in [project tasks](https://docs.platformio.org/en/latest/projectconf/sections/env/options/build/build_unflags.html) -> Advanced -> Verbose Build.
